language/src/regent/optimize_index_launches.t

```diff
@@ -1257,21 +1257,26 @@ local function collapse_projection_functor(pf, dim, bounds)
 end
 
 local function insert_dynamic_check(args_need_dynamic_check, index_launch_ast, unopt_loop_ast)
-  local stats = terralib.newlist()
+  local check = terralib.newlist()
+
+  local index_types = { [0] = std.int1d, std.int1d, std.int2d, std.int3d }
+  local rect_types = { [0] = std.rect1d, std.rect1d, std.rect2d, std.rect3d }
+
+  local is_demand = unopt_loop_ast.annotations.index_launch:is(ast.annotation.Demand)
```

Comment:
Note: this needs to be slightly more general, because it can also be triggered by `__demand(__constant_time_launch)` (which implies `__demand(__index_launch)`). Perhaps, instead of computing it here, pass it in from the place where `report_fail` is assigned?

Reply:
Pushed.